nrf5/hal/hal_timer.c

```diff
@@ -26,41 +26,76 @@
 
 #include "mphalport.h"
 #include "hal_timer.h"
+#include "hal_irq.h"
 
 #ifdef HAL_TIMER_MODULE_ENABLED
 
+static hal_timer_app_callback m_callback;
+
+void hal_timer_callback_set(hal_timer_app_callback callback) {
+    m_callback = callback;
+}
+
 void hal_timer_init(hal_timer_conf_t const * p_timer_conf) {
+    NRF_TIMER_Type * p_timer = TIMER_BASE(p_timer_conf->id);
+
+    p_timer->CC[0]       = 1000 * p_timer_conf->period;
+    p_timer->MODE        = TIMER_MODE_MODE_Timer;
+    p_timer->BITMODE     = TIMER_BITMODE_BITMODE_24Bit << TIMER_BITMODE_BITMODE_Pos;
+    p_timer->PRESCALER   = 4; // 1 us
+    p_timer->INTENSET    = TIMER_INTENSET_COMPARE0_Msk;
+    p_timer->SHORTS      = (TIMER_SHORTS_COMPARE0_CLEAR_Enabled << TIMER_SHORTS_COMPARE0_CLEAR_Pos);
+    p_timer->TASKS_CLEAR = 1;
 
+    hal_irq_priority(TIMER_IRQ_NUM(p_timer_conf->id), 3);
```

Comment:
3?
shouldn't it be `hal_irq_priority(TIMER_IRQ_NUM(p_timer_conf->id), p_timer_conf->irq_priority);` ?